api/app/auto_spatial_advisory/hfi_percent_conifer.py

```diff
@@ -0,0 +1,161 @@
+    with gdal.Open(pct_conifer_key) as conifer_ds, gdal.Open(hfi_key) as hfi_ds:
+        if not rasters_match(conifer_ds, hfi_ds):
+            logger.error(f"{pct_conifer_key} and {hfi_key} do not match.")
+            return
+        for zone in zone_units:
+            zone_wkb = zone.geom
+            shapely_geom = to_shape(zone_wkb)
+            zone_wkt = shapely_geom.wkt
+            zone_geom = prepare_wkt_geom_for_gdal(zone_wkt, source_srs)
+
+            warp_options = gdal.WarpOptions(cutlineWKT=zone_geom, cutlineSRS=zone_geom.GetSpatialReference(), cropToCutline=True)
+
+            conifer_path = "/vsimem/percent_conifer.tif"
+            intersected_ds: gdal.Dataset = gdal.Warp(conifer_path, conifer_ds, options=warp_options)
+            pct_conifer_clip = intersected_ds.GetRasterBand(1).ReadAsArray()
+
+            hfi_path = "/vsimem/zone_hfi.tif"
+            intersected_ds: gdal.Dataset = gdal.Warp(hfi_path, hfi_ds, options=warp_options)
+            hfi_array_clip = intersected_ds.GetRasterBand(1).ReadAsArray()
+
+            # make sure the previous in-memory files are deleted before the next loop
+            gdal.Unlink(conifer_path)
+            gdal.Unlink(hfi_path)
+
+            min_pct_conifer = get_minimum_percent_conifer_for_hfi(pct_conifer_clip, hfi_array_clip)
+
+            if min_pct_conifer:
+                record = AdvisoryHFIPercentConifer(
+                    advisory_shape_id=zone.id,
+                    fuel_type=mixed_fuel_record.id,
+                    run_parameters=run_parameters_id,
+                    min_percent_conifer=int(min_pct_conifer),
+                )
+                all_hfi_conifer_percent_to_save.append(record)
+
+    if len(all_hfi_conifer_percent_to_save) > 0:
+        await save_all_percent_conifer(session, all_hfi_conifer_percent_to_save)
+
+
+def get_minimum_percent_conifer_for_hfi(pct_conifer_array: np.ndarray, hfi_array: np.ndarray) -> float:
+    """
+    Extracts the minimum percent conifer given an array of hfi data and a threshold.
+    The percent conifer grid only has values above 0 where there is M-1/M-2 fuel type in the fuel grid.
+
+    :param pct_conifer_array: Array of wind speed values extracted from raster
+    :param hfi_array: Array of hfi values extracted from raster
+    :return: minimum percent conifer
+    """
+    mask = (hfi_array > 4000) & (pct_conifer_array > 0)
+    min_pct_conifer = np.nanmin(pct_conifer_array[mask]) if np.any(mask) else None
+
+    return min_pct_conifer
+
+
+async def save_all_percent_conifer(session: AsyncSession, hfi_min_percent_conifer: list[AdvisoryHFIPercentConifer]):
+    logger.info("Writing HFI Advisory Minimum Percent Conifer")
+    session.add_all(hfi_min_percent_conifer)
+
+
+async def start_hfi_percent_conifer(args: argparse.Namespace):
+    async with get_async_write_session_scope() as db_session:
+        run_parameters = await get_run_parameters_by_id(db_session, int(args.run_parameters_id))
```

Comment:
Can we destructure `run_parameters` into `run_type`, `run_datetime`, `for_date` so we don't have to index into the single result?